bundles/org.openhab.ui/web/src/assets/definitions/blockly/ohblocks_timers.js

```diff
@@ -0,0 +1,326 @@
+/*
+* This code has been originally provided by https://github.com/bigbasec
+* @author stefan.hoehn several changes to make the provided code working
+*
+*/
+import Blockly from 'blockly'
+import { FieldItemModelPicker } from './ohitemfield'
+
+export default function defineOHBlocks_Timers (f7) {
+  /*
+  * Field to enter the timer name
+  * Blockly part
+  */
+  Blockly.Blocks['oh_timer_item'] = {
+    init: function () {
+      this.appendDummyInput()
+        .appendField('Timer Name')
+        .appendField(new Blockly.FieldTextInput('MyTimer'), 'timerName')
+      this.setColour(0)
+      this.setInputsInline(true)
+      this.setTooltip('Timer name selection')
+      this.setOutput(true, null)
+    }
+  }
+
+  /*
+  * Field to enter the timer name
+  * Blockly part
+  */
+  Blockly.JavaScript['oh_timer_item'] = function (block) {
+    const timerName = block.getFieldValue('timerName')
+    let code = timerName
+    return [code, 0]
+  }
+
+  /*
+  * sleeps for the number of milliseconds
+  * Blockly part
+  */
+  Blockly.Blocks['oh_sleep'] = {
+    init: function () {
+      this.appendDummyInput()
+        .appendField('Thread Sleep (ms)')
+        .appendField(new Blockly.FieldNumber(1000), 'milliseconds')
+      this.setPreviousStatement(true, null)
+      this.setNextStatement(true, null)
+      this.setColour(230)
+      this.setTooltip('Waits for the specified milliseconds')
+    }
+  }
+
+  /*
+  * sleeps for the number of milliseconds
+  * Code part using Thread.sleep
+  */
+  Blockly.JavaScript['oh_sleep'] = function (block) {
+    const voiceName = Blockly.JavaScript.provideFunction_(
+      'thread',
+      ['var ' + Blockly.JavaScript.FUNCTION_NAME_PLACEHOLDER_ + ' = Java.type(\'java.lang.Thread\')'])
+    let milliseconds = block.getFieldValue('milliseconds')
+
+    let code = `thread.sleep(${milliseconds})\n`
+    return code
+  }
+
+  /*
+  * Simple Timer creation without further control of retriggering, canceling or the like
+  * Blockly part
+  */
+  Blockly.Blocks['oh_simpleTimer'] = {
+    init: function () {
+      this.appendStatementInput('timer')
+        .setCheck(null)
+        .appendField('After')
```

Comment:
after

Make sure all labels in Blockly are lower case and read like a natural languagge sentence (i.e. no technical function names etc., see the other existing blocks for inspiration).

Reply:
Sure, found some more.